src/couch_views/src/couch_views_fdb.erl

```diff
@@ -352,6 +378,28 @@ process_rows(Rows) ->
 
 
 calculate_row_size(Rows) ->
+    KeyLimit = key_size_limit(),
+    ValLimit = value_size_limit(),
+
     lists:foldl(fun({K, V}, Acc) ->
-        Acc + erlang:external_size(K) + erlang:external_size(V)
+        KeySize = erlang:external_size(K),
+        ValSize = erlang:external_size(V),
+
+        if KeySize =< KeyLimit -> ok; true ->
+            throw({size_exceeded, key})
+        end,
+
+        if ValSize =< ValLimit -> ok; true ->
+            throw({size_exceeded, value})
+        end,
+
+        Acc + KeySize + ValSize
     end, 0, Rows).
+
+
+key_size_limit() ->
+    config:get_integer("couch_views", "key_size_limit", 8000).
```

Comment:
Let's define 8000 and 64000 as constants at the top.

Also what do you think about actually clipping the user provided config to those maximums? 

`max(config:get_integer(...), ?MAX_KEY_SIZE_LIMIT)` ?